### PR TITLE
[WIP] Apollo caching

### PR DESCRIPTION
**Pull request by leviadam**
The idea of this PR is to use the cache in order to reduce the number of calls to the subgraph.

### CODE REVIEW

File: src/dao.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export class DAO implements IStateful<IDAOState> {
         nativeReputation { id, totalSupply }
         nativeToken { id, name, symbol, totalSupply }
         membersCount
-        members (where: {address:"${this.address}"}) {
+        members {
```
Comment (jellegerbrandy) [inline src/dao.ts]: wait why? We are getting the totken and rep balance of the dao here.

File: src/arc.ts [hunk 2]
```diff
@@ -96,6 +96,14 @@ export class Arc {
       {
         daos {
           id
+          name
```
Comment (jellegerbrandy) [inline src/arc.ts]: why?

File: src/arc.ts [hunk 3]
```diff
@@ -173,7 +181,7 @@ export class Arc {
       Logger.debug(query.loc.source.body)
 
       if (!apolloQueryOptions.fetchPolicy) {
-        apolloQueryOptions.fetchPolicy = 'network-only'
+        apolloQueryOptions.fetchPolicy = 'cache-first'
```
Comment (jellegerbrandy) [inline src/arc.ts]: yes, indeed.

File: src/utils.ts [hunk 4]
```diff
@@ -65,7 +65,17 @@ export function createApolloClient(options: {
   //     wsorhttplink
   //   ])
   const client = new ApolloClient({
-    cache: new InMemoryCache(),
+    cache: new InMemoryCache({
+      cacheRedirects: {
+        Query: {
+          dao: (_, args, { getCacheKey }) =>  getCacheKey({ __typename: 'DAO', id: args.id }),
```
Comment (jellegerbrandy) [inline src/utils.ts]: why is this needed?

Comment (leviadam) [reply src/utils.ts]: This is how apollo cache works.
If you call a list say daos[], and later call a specific dao, it will not get it from the cache, unless you do a cache Redirect. Not sure why, I was surprised by it, but that is what I found in the documentation.

File: src/arc.ts (hunk 2 above)
Comment (leviadam) [reply src/arc.ts]: I wanted to cache it so it will be available for the later calls.

File: src/dao.ts (hunk 1 above)
Comment (leviadam) [reply src/dao.ts]: I thought it is done automatically. Maybe I'm wrong.

**Comment by jellegerbrandy** [comment]
Thanks adam. Most of these changes are in #222, so I'm closing this...
